src/org/sosy_lab/java_smt/example/NQueens.java

```diff
@@ -50,36 +46,42 @@
 import org.sosy_lab.java_smt.api.SolverException;
 public class NQueens {
     public static void main(String... args)
-            throws InvalidConfigurationException, SolverException, InterruptedException, IOException {
+        throws InvalidConfigurationException, SolverException, InterruptedException, IOException {
         Configuration config = Configuration.defaultConfiguration();
         LogManager logger = BasicLogManager.create(config);
-        ShutdownNotifier notifier = ShutdownNotifier.createDummy();{
-            Solvers solver = Solvers.SMTINTERPOL;
-            try (SolverContext context =
-                         SolverContextFactory.createSolverContext(config, logger, notifier, solver)) {
-                NQueensSolver MyQueen = new NQueen(context,4);
-                Boolean[][] solution = MyQueen.solve(4);
-                if (solution == null) {
-                    System.out.println("The Queen can't be placed in this condition.");
+        ShutdownNotifier notifier = ShutdownNotifier.createDummy();
+        Solvers solver = Solvers.PRINCESS;
+        try (SolverContext context = SolverContextFactory.createSolverContext(config, logger, notifier, solver)) {
+            /*
+            The outer loop present here is used to check whether my constraints give out correct
+            solution for any of the value on n upto 12
+            */
+            for (int n = 1; n <= 12; n++) {
+                NQueensSolver MyQueen = new NQueen(context, n);
+                Boolean[][] solutions = MyQueen.solve(n);
+                if (solutions==null) {
+                    System.out.println("No solutions found for " + n + " queens.");
                 } else {
-                    for (int row = 0; row < solution.length; row++) {
-                        for (int col = 0; col < solution[0].length; col++) {
-                            if (solution[row][col]) {
-                                System.out.print("Q ");
-                            } else {
-                                System.out.print("_ ");
+                    System.out.println("Solutions for " + n + " queens:");
+                    for (Boolean[] sol : solutions) {
```

Comment:
The variable "sol" is unused, and the CI even warns about it. Please improve this.